### ISSUE TITLE
Does F3 work inside a macro?

### ISSUE THREAD

**Issue by Ypot** [Silex/elmacro#30]
Something is wrong.
I got some time to try elmacro again, but if I use F3 inside a macro, it always shows "last-command-event" variable:

(defun last-macro2 ()
  (interactive)
  (setq last-command-event 13)
  (org-return nil nil 1)
  (setq last-command-event 13))

**Comment by Silex** [comment]
Good catch, it's because it calls `kmacro-start-macro-or-insert-counter` so I only take from the last one to `kmacro-end-macro`. I need to take the first one, but only starting from the end of the macro.

I need to think of a resilient algorithm to do that.
